### PR TITLE
Fixes:[T348192] Add max character limit while creating identifier in Internet Archive and remove some special characters

### PR DESCRIPTION
**Pull request by okerekechinweotito**
Fixes: [T348192](https://phabricator.wikimedia.org/T348192)

## Proposed Changes
- Validates if generated identifier is less than 50 characters and alphanumeric
- Prompts user to enter a new identifier if validation fails

## Files Created/Updated

- components/ChangeIdentifier.js - Extract the component used in handling duplicate identifier into a new component called ChangeIdentifier to enable efficient reuse
- components/Books.js - add a conditional that checks if the initial generated identifier is Alphanumeric and not more than 50 characters.  If it fails the validation , reuse the new ChangeIdentifier component and prompt the user to enter a new identifier that meets the requirements.

## Current Look

Screenshot (27)

## Checklist 
- [x] Coding Conventions are followed.
- [x] Comments are used for Documenting the Code.
- [x] Correct File Names are mentioned.

### CODE REVIEW

**Comment by okerekechinweotito** [comment]
@Parthiv-M Good-day
Please does this meet the criteria for [T348192](https://phabricator.wikimedia.org/T348192)

**Comment by Parthiv-M** [comment]
Works fine for me

File: components/Books.js [hunk 1]
```diff
@@ -379,53 +403,59 @@ class Books extends React.Component {
...
+                inputPlaceholder="Enter unique file identifier"
+                onIdentifierChange={(event) =>
+                  this.setState({ IAIdentifier: event.target.value })
+                }
+              />
             ) : null}
+
+            {this.state.isValidIdentifier === false ? (
+              <ChangeIdentifier
+                description={
```
Comment (coderwassananmol) [inline components/Books.js:428]: Please change the description to:

"The file you want to upload with title - {" "} {this.state.IATitle} either contains special characters or exceeds 50 characters in length. Please provide an identifier that consists only of letters (A-Z) and numbers (0-9)."

Comment (okerekechinweotito) [reply components/Books.js:428]: @coderwassananmol 
I have corrected the description

**Review by coderwassananmol — APPROVED** [review]
LGTM

**Pull request merged**